exporter/sapmexporter/exporter.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package sapmexporter exports trace data using Splunk's SAPM protocol.
+package sapmexporter
+
+import (
+	"context"
+	"errors"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumererror"
+	"github.com/open-telemetry/opentelemetry-collector/exporter"
+	"github.com/open-telemetry/opentelemetry-collector/exporter/exporterhelper"
+	"github.com/open-telemetry/opentelemetry-collector/translator/trace/jaeger"
+	sapmclient "github.com/signalfx/sapm-proto/client"
+	"go.uber.org/zap"
+)
+
+// sapmExporter is a wrapper struct of SAPM exporter
+type sapmExporter struct {
+	client *sapmclient.Client
+	logger *zap.Logger
+}
+
+func (*sapmExporter) Name() string {
+	return "sapm"
+}
+
+func (se *sapmExporter) Shutdown() error {
+	se.client.Stop()
```

Comment:
Q.: What happens with any flying request? Does it wait for them to finish?

Reply:
Yes, it waits for them to finish and if more requests come in while it is waiting then it waits for them as well. In practice it can accept <N more requests after Stop() is called and finish them before Stop() returns where N is number of workers.